hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,355 @@
+            # replace the frame number with frame spec
+            seq_path = publisher.util.get_frame_sequence_path(path)
+            if seq_path:
+                evaluated_path = seq_path
+                type_display = "%s Sequence" % (type_display,)
+                item_type = "%s.%s" % (item_type, "sequence")
+                is_sequence = True
+
+        display_name = publisher.util.get_publish_name(
+            path, sequence=is_sequence)
+
+        # create and populate the item
+        file_item = parent_item.create_item(
+            item_type, type_display, display_name)
+        file_item.set_icon_from_path(item_info["icon_path"])
+
+        # if the supplied path is an image, use the path as # the thumbnail.
+        if (item_type.startswith("file.image") or
+            item_type.startswith("file.texture")):
+            file_item.set_thumbnail_from_path(path)
+
+            # disable thumbnail creation since we get it for free
+            file_item.thumbnail_enabled = False
+
+        # all we know about the file is its path. set the path in its
+        # properties for the plugins to use for processing.
+        file_item.properties["path"] = evaluated_path
+
+        if is_sequence:
+            # include an indicator that this is an image sequence and the known
+            # file that belongs to this sequence
+            file_item.properties["is_sequence"] = True
+            file_item.properties["sequence_files"] = [path]
+
+        self.logger.info("Collected file: %s" % (evaluated_path,))
+
+        return file_item
+
+    def _collect_folder(self, parent_item, folder):
+        """
+        Process the supplied folder path.
+
+        :param parent_item: parent item instance
+        :param folder: Path to analyze
+        :returns: The item that was created
+        """
+
+        # make sure the path is normalized. no trailing separator, separators
+        # are appropriate for the current os, no double separators, etc.
+        folder = sgtk.util.ShotgunPath.normalize(folder)
+
+        publisher = self.parent
+        img_sequences = publisher.util.get_frame_sequences(
+            folder, IMAGE_EXTENSIONS_LIST)
+
+        file_items = []
+
+        for (image_seq_path, img_seq_files) in img_sequences:
+
+            # get info for the extension
+            item_info = self._get_item_info(image_seq_path)
+            item_type = item_info["item_type"]
+            type_display = item_info["type_display"]
+
+            # the supplied image path is part of a sequence. alter the
+            # type info to account for this.
+            type_display = "%s Sequence" % (type_display,)
+            item_type = "%s.%s" % (item_type, "sequence")
+            icon_name = "image_sequence.png"
+
+            # get the first frame of the sequence. we'll use this for the
+            # thumbnail and to generate the display name
+            img_seq_files.sort()
+            first_frame_file = img_seq_files[0]
```

Comment:
In a lot of cases, the first frame will be inside of an element's handle and won't be a representative image. Should we take the median frame instead?